### ISSUE TITLE
What should the LICENSE file in each repository contain?

### ISSUE THREAD

**Issue by domenic** [whatwg/sg#51]
> Notice for Repositories
>
> The repositories for all WHATWG publications and other deliverables must include the text of the applicable license in a LICENSE file.

I went searching for how to do this, exactly, and it's not entirely clear. See e.g. https://wiki.creativecommons.org/wiki/Marking_your_work_with_a_CC_license which seems to recommend that for web pages just the notice in the footer is desirable, or https://creativecommons.org/2014/01/07/plaintext-versions-of-creative-commons-4-0-licenses/ which says "linking directly to the deeds (say with the copy-paste output you get with the license chooser) is good enough, even ideal".

Worse, the link at https://creativecommons.org/2014/01/07/plaintext-versions-of-creative-commons-4-0-licenses/ , to https://creativecommons.org/licenses/by/4.0/legalcode.txt , 404s.

I see a few options:

- Try to hunt down a plaintext or Markdown version of the license to include in the LICENSE file. If the SG prefers this route, I'd appreciate help hunting one down.
- Make the LICENSE file include a pointer to the appropriate location, somewhat like those above resources are suggesting. E.g., the entirety of the file is just:
  ```
  This work is licensed under a Creative Commons Attribution 4.0
  International License: http://creativecommons.org/licenses/by-sa/4.0/.
  ```
- Remove the provision to include a LICENSE file.

What would the SG prefer?

**Comment by annevk** [comment]
I was thinking we'd copy https://github.com/whatwg/sg/blob/master/LICENSE.md as-is. This is a change in filename, but that seems acceptable.

**Comment by annevk** [comment]
Although what is weird is that GitHub doesn't recognize it as CC BY (https://github.com/whatwg/xhr on the right says CC0-1.0).

**Comment by domenic** [comment]
Huh, I wonder where that file came from.

**Comment by domenic** [comment]
I will tentatively go with option 1, based on that file that @annevk found.

I think GitHub doesn't recognize it because https://github.com/github/choosealicense.com/issues/33. Not much we can do about that besides maybe asking GitHub folks more directly.

**Comment by annevk** [comment]
Should we use https://github.com/creativecommons/legalcode/blob/master/by-nc-sa_4.0.txt by the way? Or does CC also host a Markdown variant somewhere that @othermaciej ended up adding and I referenced above (with changes for our specific copyright statement)?

**Comment by domenic** [comment]
I don't think we should use By-NC-SA, but I guess https://github.com/creativecommons/legalcode/blob/master/by_4.0.txt is what you meant.

**Comment by domenic** [comment]
Another minor question is, if we go with the plaintext version linked to above, whether the filename is LICENSE vs. LICENSE.txt. I mildly prefer .txt, but will assume no extension unless told otherwise.

**Comment by annevk** [comment]
No rename is nice as it's one more cool URL that way.

**Comment by domenic** [comment]
Also I discovered (https://github.com/github/choosealicense.com/issues/33#issuecomment-352811793 for more details) that if you use the plaintext version GitHub will auto-recognize it and display it as such. So I guess I'm leaning that way now.

**Comment by othermaciej** [comment]
I think LICENSE.md in this repo is suitable, but if GitHub doesn’t know how to auto recognize it perhaps we should switch to plaintext.

**Comment by othermaciej** [comment]
The contents of the LICENSE file at https://github.com/domenic/test-license-stuff/tree/master look fine to me, and we should probably switch this repository from Markdown to plaintext, to enable GitHub's license auto-recognition. LICENSE.txt would be slightly nicer if GitHub recognizes it, but failing that LICENSE seems fine.

**Comment by mlinksva** [comment]
The markdown headers in https://github.com/whatwg/sg/blob/eab04b1dd27b15b161fab5e1d01f0ac68552242f/LICENSE.md are probably the problem. There's an issue for that in https://github.com/benbalter/licensee/issues/247 ... but switching to an unadorned copy of the license should fix license detection for this repo.

**Comment by geoffcr** [comment]
Thanks for catching this - license detection is a significant consideration, and a .txt file is going to be the most readily accessible to most engines.

**Comment by domenic** [comment]
It seems we've settled on the plaintext license file, as seen e.g. in https://github.com/whatwg/url/pull/360/files. If anyone disagrees, especially anyone in @whatwg/sg, please let me know by tomorrow if possible. I plan to spend some time tomorrow sending updates to all WHATWG specs at once, and I'd ideally like to not have to re-do them :).

**Comment by othermaciej** [comment]
I agree with the approach of plaintext license in a file named LICENSE. Assuming other SG members sign off, is this issue sufficient, or would anyone like to have this clarified in Workstream Policy? Right now it says:

> The repositories for all WHATWG publications and other deliverables must include the text of the applicable license in a LICENSE file.

**Comment by annevk** [comment]
I think the only annoying bit is that the CC BY plain text licence is preceded by

> Copyright © 2017 WHATWG (Apple, Google, Mozilla, Microsoft).
> This work is licensed under a Creative Commons Attribution 4.0 International License:
>

meaning we'll have to update it yearly.

**Comment by domenic** [comment]
I think the Workstream policy could be clarified, perhaps by saying something like "identical to the one in this repository" or similar, after #53 gets merged.

**Comment by domenic** [comment]
> I plan to spend some time tomorrow sending updates to all WHATWG specs at once, and I'd ideally like to not have to re-do them :).

Update: given how many people are headed off into their vacations, and the fact that I'd have to update everything for the 2018 copyright year change anyway, I've decided to hold off on all these updates until the new year. Hopefully that's OK with all.

**Comment by othermaciej** [comment]
@domenic Seems ok to wait for 2018. 

To respond to @annevk's points:

There are two obvious alternatives to having a copyright statement with a date in the LICENSE file:
1. No copyright notice at the top at all.
2. Copyright notice, but without a year.

On (1): almost all the popular projects on GitHub include a copyright notice in their LICENSE file. Based on this apparent norm, I'd be very reluctant to omit it without prior legal review of whether it is ok. And I'd much rather be like other projects without a very compelling reason not to. On an informal level, it seems a bit strange to state the license without telling the reader who is granting them that license.

On (2): I have heard before that it's never correct to have a copyright notice without a date, but it does seem some popular open source projects do this in their LICENSE files. If we wanted to do this, we should get legal review of this idea first.

In conclusion, the safest short term option is to keep the copyright notice with date. If it would be a large time savings to avoid having a date that needs updating in the LICENSE file, we can consult counsel about the alternatives. Or we could figure out a way to automate the process.

**Comment by mlinksva** [comment]
> almost all the popular projects on GitHub include a copyright notice in their LICENSE file

That's only the case for "template" licenses such as MIT or BSD-*. Others such GPL, MPL, EPL, Apache, and CC licenses aren't, and so are typically completely deployed as LICENSE files verbatim, unmodified. Copyright notices go elsewhere such as in README or NOTICES files, source file comments, rendered documents, depending on the context.

If you want to be certain that your LICENSE file will be detected, it's best to not add anything to non-templated licenses. Apologies, I failed to notice the copyright/license notice at the top when looking at this previously!

**Comment by geoffcr** [comment]
IMHO, we don't need the year; it can just say Copyright (C) WHATWG etc.
Tagging others in case there are different viewpoints: @vfournier17, @ishnaneamatullah, @eleemoz, (unable to tag hpworkman for some reason)

**Comment by vfournier17** [comment]
What does the Copyright Act require?  I admittedly haven't looked at it recently.

**Comment by js-choi** [comment]
TL;DR: I’m not a lawyer, but the US’s answer seems to be: You don’t need a copyright notice on works, but if you do decide to include one in a written standard, it should have a date, it should be updated, but the notice is merely informative either way, [except to prevent innocent-infringement defenses during litigation](https://www.copyright.gov/title17/92chap4.html).

Gory details:

The [US Copyright Act of 1976](//en.wikipedia.org/wiki/Copyright_Act_of_1976) makes up the bulk of current US copyright law, and it requires that all works with a few exceptions have copyright notices including years. Exceptions included “postcards, stationary, jewelry, dolls, toys, or any useful article”. But [when the US joined the Berne Convention in 1988](//en.wikipedia.org/wiki/Berne_Convention_Implementation_Act_of_1988), the law had to be amended such that copyright notices became implied and completely optional for works created since 1989-03-01.

However, the US Copyright Office strongly encourages copyright owners to use a notice even post Berne. The [US Copyright Office’s 2017 Circular 3](//www.copyright.gov/circs/circ03.pdf) lists several benefits including making “potential users aware that copyright is claimed in the work”, preventing “a defendant…from attempting to limit…liability…based on an innocent infringement defense”, identifying “the copyright owner at the time the work was first published for parties seeking permission to use the work”, and preventing “the work from becoming an orphan work”.

The [Compendium of U.S. Copyright Office Practices 2017, Chapter 2200](//www.copyright.gov/comp3/chap2200/ch2200-notice.pdf) clarifies further:

> Copyright owners may use any form of notice for an unpublished work, a foreign work, or a work published on or after March 1, 1989. As a general rule, Post-Berne works do not need to comply with the notice requirements set forth in Sections 401 or 402 of the Copyright Act or any of the other requirements discussed in Sections 2203 through 2209 below.

> Note: There is limited exception to this rule. Works published on or after March 1, 1989 may require a notice that complies with Sections 401 or 402 to prevent a defendant from invoking an innocent infringement defense in a copyright infringement action. See 17 U.S.C. §[§ 401(d)](https://www.copyright.gov/title17/92chap4.html),
402(d), 504(c)(2).

**Comment by dbaron** [comment]
Plaintext license file called `LICENSE` sounds good to me as well.

**Comment by othermaciej** [comment]
@henceproved @michaelchampion I'd appreciate indications of agreement (or disagreement). The proposal is to have a plaintext LICENSE file in each repository just like the one in the SG repo, and to spell that out in Workstream Policy. See the Pull Request for details.

**Comment by annevk** [comment]
Note that @domenic and I already implemented the LICENSE change for all repositories. I was somehow under the impression we already had agreement here. See https://github.com/whatwg/fetch for an example. (This can be changed in case of disagreement though.)

**Comment by othermaciej** [comment]
What was actually done in all the repos is fine, since the current policy is not at all specific about what goes in the LICENSE file. What I'm asking for SG agreement on is to make the Policy actually spell out a specific option, which happens to match what was already done, for future reference.

**Comment by henceproved** [comment]
Sounds good to me.
